### PR TITLE
More configurable HAProxy Settings

### PR DESCRIPTION
**Pull request by Grobix**
Now all tests pass

### CODE REVIEW

File: templates/haproxy.cfg.j2 [hunk 1]
```diff
@@ -12,40 +12,68 @@ global
...
 {% else %}
-        timeout connect 5000
-        timeout client 50000
-        timeout server 50000
+    timeout connect 5000
+    timeout client 50000
+    timeout server 50000
 {% endif %}
 {% if ansible_os_family == 'Debian' %}
-  errorfile 400 /etc/haproxy/errors/400.http
```
Comment (kostyrev) [inline templates/haproxy.cfg.j2:29]: I fail to see the difference.
Help me out?

Comment (geerlingguy) [inline templates/haproxy.cfg.j2:29]: Looks like @Grobix added extra spaces in the YAML; this needs to be fixed so the PR doesn't make whitespace inconsistent.

**Comment by thebergamo** [comment]
Probably this PR can solve my problem! 
The cookie check are put me in troubles. 
`server {{ backend.name }} {{ backend.address }} cookie {{ backend.name }} check` actually I cannot disable this. But this PR empower me to disable this check!

**Comment by thebergamo** [comment]
@geerlingguy some news about this PR?

**Comment by geerlingguy** [comment]
Unfortunately, I'm not a fan of the variable structure this introduces (e.g. removing `haproxy_backend_servers` and a bunch of other `haproxy_*` variables and replacing with a highly structured `haproxy_config` dict.

I would rather consider first adding in any particular configurable variables (with names like `haproxy_[variable]`), then later in a separate PR consider any more rearchitecture (since that would necessitate a major version bump since I don't want to break everyone's existing configs).

If you need to add arbitrary additional config lines per backend_server, or do things like that, we could consider doing something similar to `extra_parameters` in my `apache` role here: https://github.com/geerlingguy/ansible-role-apache/blob/master/templates/vhosts.conf.j2#L29-L31

**Pull request closed without merging**